### ISSUE TITLE
MsalExtensionException constructor always throws ArgumentNullException

### ISSUE THREAD

**Issue by FreddyD-GH** [AzureAD/microsoft-authentication-library-for-dotnet#1947]
**Which Version of MSAL are you using ?**
4.16.0

**Platform**
.NET Core 3.1

**What authentication flow has the issue?**
* Desktop / Mobile
    * [x ] Interactive
    * [ ] Integrated Windows Auth
    * [ ] Username Password
    * [ ] Device code flow (browserless)
* Web App
    * [ ] Authorization code
    * [ ] OBO
* Web API
    * [ ] OBO

**Is this a new or existing app?**
This is a new app or experiment

**Repro**
Try to create a new MsalExtensionException by using the following constructor:
```csharp
public MsalExtensionException(string message, Exception innerException)
            : base(message, string.Empty, innerException)
```
No matter what you pass for arguments to the constructor, an ArgumentNullException will be thrown, because it is passing `string.Empty` as the message parameter to the base constructor, which causes any new MsalExtensionException to throw an ArgumentNullException, as you can see in the constructor for the parent class:
```csharp
public MsalException(string errorCode, string errorMessage, Exception innerException)
            : base(errorMessage, innerException)
        {
            if (string.IsNullOrWhiteSpace(Message))
            {
                throw new ArgumentNullException(nameof(errorMessage));
            }

            ErrorCode = errorCode;
        }
```
**Expected behavior**
There should be a way to not get an ArgumentNullException when using this constructor.

**Actual behavior**
An ArgumentNullException is thrown.

**Possible Solution**
Perhaps the order of the arguments being passed to the base constructor was inadvertently mixed up.  I'm not sure what the intent was here, but I would think that `base(string.Empty, message, innerException)` would probably make more sense than `base(message, string.Empty, innerException)`.

**Comment by jennyf19** [comment]
@FreddyD-GH thanks for all the details on the issue. 
cc: @trwalke @neha-bhargava

**Comment by bgavrilMS** [comment]
@FreddyD-GH - what are you trying to do ? It looks like this exception was made public by mistake, but it is not actually used by any of our extesnion points. 

I think you should be fine throwing whatever exceptions you see fit and MSAL will log and rethrow them. The only "special" exception is `OperationCanceledException` to denote cancellation https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/67325c7cabb9e4dbccc32c6764c865f088b8f3ad/src/client/Microsoft.Identity.Client/UI/CustomWebUiHandler.cs#L61

**Comment by bgavrilMS** [comment]
Since this type is unusable, I will delete it even if it is public.

**Comment by FreddyD-GH** [comment]
I don't particularly need this exception, I was only following the documentation.  For example, [the CustomWebUi documentation](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/CustomWebUi) says:

> In case of issues, your implementation should throw a MsalExtensionException exception in order to nicely cooperate with MSAL.

I think it may be mentioned one or two other places in the docs as well.

**Comment by bgavrilMS** [comment]
Ack, I'll check for docs and update.

**Comment by neha-bhargava** [comment]
Fixed in [MSAL 4.17.0 release](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/releases/tag/4.17.0)
